diff -rupN xerces-c-3.2.4/src/CMakeLists.txt xerces-c-3.2.4-new/src/CMakeLists.txt
--- xerces-c-3.2.4/src/CMakeLists.txt	2022-10-05 22:41:35.000000000 +0200
+++ xerces-c-3.2.4-new/src/CMakeLists.txt	2022-12-05 13:57:51.402873420 +0100
@@ -1284,6 +1284,8 @@ if(MSVC)
   set_target_properties(xerces-c PROPERTIES OUTPUT_NAME "xerces-c_${INTERFACE_VERSION_MAJOR}")
   set_target_properties(xerces-c PROPERTIES RUNTIME_OUTPUT_NAME "xerces-c_${INTERFACE_VERSION_U}")
   set_target_properties(xerces-c PROPERTIES DEBUG_POSTFIX "D")
+elseif(MINGW)
+  set_target_properties(xerces-c PROPERTIES SUFFIX "-${INTERFACE_VERSION_D}${CMAKE_SHARED_LIBRARY_SUFFIX}")
 elseif(UNIX)
   # For strict libtool compatibility on Unix.  It's a horrible hack to
   # set the version in the filename, and create the symlink at install
@@ -1310,7 +1312,7 @@ else()
 endif()
 set_target_properties(xerces-c PROPERTIES FOLDER "Library")
 
-if(WIN32)
+if(MSVC)
   set(xerces_config_dir "cmake")
 else()
   set(xerces_config_dir "${CMAKE_INSTALL_LIBDIR}/cmake/XercesC")
diff -rupN xerces-c-3.2.4/tests/src/ThreadTest/ThreadTest.cpp xerces-c-3.2.4-new/tests/src/ThreadTest/ThreadTest.cpp
--- xerces-c-3.2.4/tests/src/ThreadTest/ThreadTest.cpp	2022-10-10 18:01:08.000000000 +0200
+++ xerces-c-3.2.4-new/tests/src/ThreadTest/ThreadTest.cpp	2022-12-05 13:44:39.956120229 +0100
@@ -118,7 +118,7 @@ void ThreadFuncs::startThread(ThreadFunc
 //
 //------------------------------------------------------------------------------
 
-#include <Windows.h>
+#include <windows.h>
 #include <process.h>
 
 typedef DWORD (WINAPI *ThreadFunc)(void *);
